src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,162 @@
+            @Parameter(name = "completionDate", description = "date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam(value = "completionDate", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime completionDate)
+            throws JsonProcessingException {
+        // For an explanation of @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+        // See: https://www.baeldung.com/spring-date-parameters
+
+        log.info("localDateTime={}", submissionDate);
+        log.info("localDateTime={}", completionDate);
+
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+        recommendationRequest.setProfessorName(professorName);
+        recommendationRequest.setProfessorEmail(professorEmail);
+        recommendationRequest.setRequesterName(requesterName);
+        recommendationRequest.setRecommendationTypes(recommendationTypes);
+        recommendationRequest.setDetails(details);
+        recommendationRequest.setStatus(status);
+        recommendationRequest.setSubmissionDate(submissionDate);
+        recommendationRequest.setCompletionDate(completionDate);
+
+        RecommendationRequest savedRecommendationRequest = recommendationRequestRepository.save(recommendationRequest);
+
+        return savedRecommendationRequest;
+    }
+
+    /**
+     * Delete a RecommendationRequest
+     * 
+     * @param id the id of the RecommendationRequest to delete
+     * @return a message indicating that the RecommendationRequest was deleted
+     */
+    @Operation(summary = "Delete a RecommendationRequest")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @DeleteMapping("")
+    public Object deleteRecommendationRequest(
+            @Parameter(name = "id") @RequestParam Long id) {
+        RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequestRepository.delete(recommendationRequest);
+        return genericMessage("RecommendationRequest with id %s deleted".formatted(id));
+    }
+
+    /**
+     * Update a single Recommendation Request
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "Update a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("")
+    public RecommendationRequest updateRecommendationRequest(
+            @Parameter(name = "id") @RequestParam Long id,
+            @RequestBody @Valid RecommendationRequest incoming) {
+
+        RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
```

Comment:
Similarly, you need to prevent users from editing one another's requests.  Associate the requestor id or requested email with the request, and then check currentUser, and make sure that the record belongs to the currentUser before doing anything destructive (edit, delete, etc.)